rts/Game/Camera.cpp

```diff
@@ -773,6 +775,47 @@ float3 CCamera::GetMoveVectorFromState(bool fromKeyState) const
 	return v;
 }
 
+bool CCamera::TracePointToMaxAltitude(const float3& point, const float rayLength, const float maxAltitude, float3& result) const
+{
+	const float3 dir = (point-pos).Normalize();
+	float dist = CGround::LinePlaneCol(pos, dir, rayLength, maxAltitude);
+	if (dist > 0.0) {
+		result = pos + dir*dist;
+		return true;
+	}
+	return false;
+}
+
+float3 CCamera::NearTheaterIntersection(const float3& dir, const float rayLength) const
+{
+	float maxAltitude = std::max(unitHandler.MaxUnitAltitude(), readMap->GetCurrMaxHeight());
+	maxAltitude = std::max(maxAltitude, featureHandler.MaxFeatureAltitude());
```

Comment:
Both the `unitHandler` max and the `featureHandler` max are initialized with `readMap->GetCurrMaxHeight()`, does that mean it's redundant here? (possibly with `assert`s or comments to document that assumption)
```suggestion
	const float maxAltitude = std::max(unitHandler.MaxUnitAltitude(), featureHandler.MaxFeatureAltitude());
```

Reply:
Yes it is, I'll remove it from here.

Reply:
done